include/gamepad/controller.hpp

```diff
@@ -53,22 +54,61 @@ class Gamepad {
          *
          */
         float operator[](pros::controller_analog_e_t joystick);
-        const Button& L1 {m_L1};
-        const Button& L2 {m_L2};
-        const Button& R1 {m_R1};
-        const Button& R2 {m_R2};
-        const Button& Up {m_Up};
-        const Button& Down {m_Down};
-        const Button& Left {m_Left};
-        const Button& Right {m_Right};
-        const Button& X {m_X};
-        const Button& B {m_B};
-        const Button& Y {m_Y};
-        const Button& A {m_A};
-        const float& LeftX = m_LeftX;
-        const float& LeftY = m_LeftY;
-        const float& RightX = m_RightX;
-        const float& RightY = m_RightY;
+
+        const Button& L1() { return m_L1; }
+
+        const Button& L2() { return m_L2; }
+
+        const Button& R1() { return m_R1; }
+
+        const Button& R2() { return m_R2; }
+
+        const Button& Up() { return m_Up; }
+
+        const Button& Down() { return m_Down; }
+
+        const Button& Left() { return m_Left; }
+
+        const Button& Right() { return m_Right; }
+
+        const Button& X() { return m_X; }
+
+        const Button& B() { return m_B; }
+
+        const Button& Y() { return m_Y; }
+
+        const Button& A() { return m_A; }
+
+        float LeftX(bool use_curve = true) {
+            if (use_curve && m_left_transformation) return m_left_transformation->get_value({m_LeftX, m_LeftY}).first;
+            else return m_LeftX;
+        }
```

Comment:
is there a specific reason why this is defined in the header file? This should be put in a source file to minimize hot package size.

Reply:
I think that since this function is included in a cold package source file, the version of the function that is included in the hot package binary should be discarded at link time, so there shouldn't be any noticeable difference in hot package size.